docs/api_examples/enrichment_analysis.py

```diff
@@ -0,0 +1,183 @@
+    .sample(
+        N_to_sample - len(idx_always_included),
+        axis=1,
+        random_state=42,
+    )
+    .join(df_omics[idx_always_included])
+)
+df_omics
+
+
+# %%
+df_meta
+
+
+# %% [markdown]
+# ## Compute up and downregulated genes
+# These will be used to find enrichments in the set of both up and downregulated genes.
+
+# %%
+group = "Status"
+covariates = ["PlatinumValue"]
+diff_reg = acore.differential_regulation.run_anova(
+    df_omics.join(df_meta[[group]]),
+    drop_cols=[],
+    subject=None,
+    group=group,
+)
+diff_reg.describe(exclude=["float"])
+
+# %%
+diff_reg["rejected"] = diff_reg["rejected"].astype(bool)  # ! needs to be fixed in anova
+diff_reg.query("rejected == True")
+
+# %% [markdown]
+# ## Find functional annotations, here pathways
+#
+
+# %%
+from acore.io.uniprot import (
+    check_id_mapping_results_ready,
+    get_id_mapping_results_link,
+    get_id_mapping_results_search,
+    submit_id_mapping,
+)
+
+
+def fetch_annotations(ids: pd.Index | list) -> pd.DataFrame:
+    """Fetch annotations for UniProt IDs. Combines several calls to the API of UniProt's
+    knowledgebase (KB).
+
+    Parameters
+    ----------
+    ids : pd.Index | list
+        Iterable of UniProt IDs. Fetches annotations as speecified by the specified fields.
+    fields : str, optional
+        Fields to fetch, by default "accession,go_p,go_c. See for availble fields:
+        https://www.uniprot.org/help/return_fields
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame with annotations of the UniProt IDs.
+    """
+    job_id = submit_id_mapping(from_db="UniProtKB_AC-ID", to_db="UniProtKB", ids=ids)
+
+    if check_id_mapping_results_ready(job_id):
+        link = get_id_mapping_results_link(job_id)
+        # add fields to the link to get more information
+        # From and Entry (accession) are the same for UniProt IDs.
+        results = get_id_mapping_results_search(
+            link + "?fields=accession,go_p,go_c,go_f&format=tsv"
+        )
+    header = results.pop(0).split("\t")
+    results = [line.split("\t") for line in results]
+    df = pd.DataFrame(results, columns=header)
+    return df
+
+
+fname_annotations = "downloaded/annotations.csv"
+fname = Path(fname_annotations)
+try:
+    annotations = pd.read_csv(fname, index_col=0)
+    print(f"Loaded annotations from {fname}")
+except FileNotFoundError:
+    print(f"Fetching annotations for {df_omics.columns.size} UniProt IDs.")
+    annotations = fetch_annotations(df_omics.columns)
+    annotations = (
+        annotations.set_index("Entry")
+        .rename_axis("identifier")
+        .drop("From", axis=1)
+        .rename_axis("source", axis=1)
+        .stack()
+        .to_frame("annotation")
+        .replace("", pd.NA)
+        .dropna()
+        .sort_values(["source", "annotation"])
+        .reset_index()
+    )
+    fname.parent.mkdir(exist_ok=True, parents=True)
+    annotations.to_csv(fname, index=True)
+
+annotations
```

Comment:
check the maximum number of genes in my dataset, explain idea of maximum proteins per pathway.

Reply:
in this example, there is a maximum of 25 proteins associated with a 'pathway' (biological annotation)